### DISCUSSION TITLE
Video format for camera eyes iCubLyon01

### DISCUSSION THREAD

**Discussion by pfdominey** [robotology/community#529] [Q&A] [answered]
Greetings,
We are consdiering using an MRI compatible stereo head-mounted display.  We want to send the video from the camera eyes to the input of thedivice.  The manufacturer wants to have some detail on the video format.  What can i tell them?
Thanks
cheers
Peter

**Comment by pattacini** [comment, marked as answer]
Hi @pfdominey 

I'm not really into the video stream things, but I can tell that YARP allows for providing in real-time the images from the cameras straight away. Thus, YARP works at the image level from point A (cameras in the eye bulbs) to point B (the laptop) and makes available images at different resolutions (320x240, 640x480...) and at different rates (e.g., 10, 15, 30 fps) depending on the hardware, the [user configuration][conf] and the bandwidth, with, optionally, different types of compressions (e.g., lossy, lossless, bayer, h264) implemented as YARP plugins.

You can't directly interface a YARP image stream with an external device because of the YARP internal protocol. 
Instead, one can develop a proxy that does the following:
1. Open a YARP port to receive the image streams.
2. If needed, process the images to do e.g. up/down-sampling, rescaling, color conversion...
3. Use the resulting images to feed the external device (the HUD) with the [protocol][protocols] required by the endpoint, over the dedicated transfer layer (e.g., UDP/TCP sockets). 

To this purpose, the proxy will need to compile against YARP and a C++ library devoted to for example video-streaming.

Given the flexibility of the YARP world, maybe the question is what are the requirements of your head-mounted display?

Let me tag other people in the know who may chime in:
@randaz81 @vtikha 

[conf]: https://github.com/robotology/robots-configuration/tree/master/iCubLyon01/camera
[protocols]: https://www.vastreaming.net/cpp-streaming-libraries.html
